samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java

```diff
@@ -0,0 +1,123 @@
+    String projectId = "YOUR-PROJECT-ID";
+    // Supported Languages: https://cloud.google.com/translate/docs/languages
+    String sourceLanguage = "your-source-language";
+    String targetLanguage = "your-target-language";
+    String inputUri = "gs://your-gcs-bucket/path/to/input/file.txt";
+    String outputUri = "gs://your-gcs-bucket/path/to/results/";
+    int timeout = 400;
+    batchTranslateDocument(projectId, sourceLanguage, targetLanguage, inputUri, outputUri, timeout);
+  }
+
+  // Batch translate document
+  public static void batchTranslateDocument(
+      String projectId,
+      String sourceLanguage,
+      String targetLanguage,
+      String inputUri,
+      String outputUri,
+      int timeout)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // refer to https://github.com/googleapis/java-translate/issues/613
+    TranslationServiceSettings.Builder translationServiceSettingsBuilder =
+        TranslationServiceSettings.newBuilder();
+    TimedRetryAlgorithm timedRetryAlgorithm =
+        OperationTimedPollAlgorithm.create(
+            RetrySettings.newBuilder().setTotalTimeout(Duration.ofSeconds(1000)).build());
+    translationServiceSettingsBuilder
+        .batchTranslateDocumentOperationSettings()
+        .setPollingAlgorithm(timedRetryAlgorithm);
+    TranslationServiceSettings translationServiceSettings =
+        translationServiceSettingsBuilder.build();
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (TranslationServiceClient client =
+        TranslationServiceClient.create(translationServiceSettings)) {
+      // The ``global`` location is not supported for batch translation
+      LocationName parent = LocationName.of(projectId, "us-central1");
+
+      // Google Cloud Storage location for the source input. This can be a single file
+      // (for example, ``gs://translation-test/input.docx``) or a wildcard
+      // (for example, ``gs://translation-test/*``).
+      // Supported file types: https://cloud.google.com/translate/docs/supported-formats
+      GcsSource gcsSource = GcsSource.newBuilder().setInputUri(inputUri).build();
+
+      BatchDocumentInputConfig batchDocumentInputConfig =
+          BatchDocumentInputConfig.newBuilder().setGcsSource(gcsSource).build();
+
+      GcsDestination gcsDestination =
+          GcsDestination.newBuilder().setOutputUriPrefix(outputUri).build();
+
+      BatchDocumentOutputConfig batchDocumentOutputConfig =
+          BatchDocumentOutputConfig.newBuilder().setGcsDestination(gcsDestination).build();
+
+      BatchTranslateDocumentRequest request =
+          BatchTranslateDocumentRequest.newBuilder()
+              .setParent(parent.toString())
```

Comment:
why doesn't this method take a LocationName?

weird that parent has two types

Reply:
Thats what I thought too.

Reply:
It looks like the [public protos are missing the `google.api.resource_reference` annotation](https://github.com/googleapis/googleapis/blob/master/google/cloud/translate/v3beta1/translation_service.proto#L1197). I have filed an internal issue for this--Googlers, see issue #196978761